### ISSUE TITLE
Crash in DBDelegate

### ISSUE THREAD

**Issue by NSStalker** [dropbox/dropbox-sdk-obj-c#334]
Hello there.

I use Your SDK 6.0.0 in my iOS app.

Here is the crash I found in FirebaseCrashlytics:

![Screenshot 2021-11-02 at 09 32 46](https://user-images.githubusercontent.com/5647878/139797491-c7cfd6f6-4068-4f91-87c9-68b6b2174ba4.png)

**Comment by greg-db** [comment]
Thanks for reporting these! Do you happen to know the steps to reproduce this? Likewise, if you can share the relevant code/parameters that the app was running at the time of the crash, that may be helpful.

Also, the latest version of the SDK is v6.2.0, which contains a few bug fixes since v6.0.0, so please update to v6.2.0 and let us know if that seems to resolve any of these. Thanks!

**Comment by NSStalker** [comment]
Yes, I anticipated this question, but unfortunately, I don't know the exact steps either, and I can only guess. I suspect the app's module is waiting for changes to the DropBox’s folders using specific cursors.
[waiting_for_db_changes.zip](https://github.com/dropbox/dropbox-sdk-obj-c/files/7462518/waiting_for_db_changes.zip)

Just in case I attached fragments of the code. The suspect module is:
**_KPDBSyncContentChangesObserver_**.

**Comment by greg-db** [comment]
Thanks!

**Comment by yoasha** [comment]
This issue also happens on my end. I see quite a lot of crashes, mostly on DBDelegate line 35 or 32, but sometimes on different lines. Seems like a memory related issue which can't be reproduced easily.
Also note that I'm using the latest version of v6.2.0
This seems to be relatively new issue as I didn't encounter it in the past (it started appearing in recent months)
I'm currently trying to investigate and gather more info. Will update here if I'll find anything.

**Comment by NSStalker** [comment]
Hello there.

Any news?

**Comment by greg-db** [comment]
@NSStalker This is still open with the team, but I don't have any update on it right now.

**Comment by yoasha** [comment]
@greg-db I know that this issue is tricky to reproduce, as I also can't reproduce it in dev environment. However, since I am using Google Crashlytics, I wonder if your team would like to get some logs to better understand the cause of this crash. You can send me detailed log requirements or even Dropbox sources files, and I can integrate them into my code. Then, once a crash is reproduced in my app, I can send you these logs for your team to analyze. We can re-iterate this process as much as needed until issue is resolved. This crash occurs almost daily on my customer devices, so there is no issue to reproduce it.

**Comment by greg-db** [comment]
@NSStalker Thanks for the note.

@yoasha Thanks for the offer! I'll bring this up with the team to see if this is something we could do.

**Comment by yoasha** [comment]
Hi,
Just following up on this issue. Please note that I still see in my crash reports many crashes in DBDelegate and wonder if you got any progress?
I realize that this issue is problematic to reproduce, as I fail to reproduce it either on my development environment, no matter which XcodeI diagnostics tools I enable. However, as I previously mentioned, these crashes are reproduced on a daily basis on my customer devices so I'm pretty sure that you can get some progress by providing me updated source code with some logs. Please strongly consider that (unless you already realize the reason) and provide a higher priority to fix these crashes (IMHO, crashes should be at the topmost priority of any software vendor).

**Comment by greg-db** [comment]
Thanks for the notes. This is still open with the team, but I don't have anything new to share unfortunately.

**Comment by yoasha** [comment]
Thanks for your quick response. Is there any chance you can discuss this with the team and check for a more precise status? For example, is anyone actively working on it?
I just wish to mention, that I use DropboxSDK for many years and encountered crashes in the past that you eventually fixed. However, the crashes of this issue appear at a much higher frequency! Please do your best to fix these crashes.

**Comment by greg-db** [comment]
@yoasha I can't make any promises myself, but yes, I'm bringing this up with the team.
